flink-core/src/main/java/org/apache/flink/util/FileUtils.java

```diff
@@ -525,6 +540,123 @@ public static Path expandDirectory(Path file, Path targetDirectory) throws IOExc
 		return new Path(targetDirectory, rootDir);
 	}
 
+	/**
+	 * List the {@code directory} recursively and return the files that satisfy the {@code fileFilter}.
+	 *
+	 * @param directory the directory to be listed
+	 * @param fileFilter a file filter
+	 * @return a collection of {@code File}s
+	 *
+	 * @throws IOException if an I/O error occurs while listing the files in the given directory
+	 */
+	public static Collection<File> listFilesInPath(final File directory, final Predicate<File> fileFilter) throws IOException {
+		checkNotNull(directory, "directory");
+		checkNotNull(fileFilter, "fileFilter");
+
+		if (!Files.exists(directory.toPath())) {
```

Comment:
Given that we always convert `File` to `nio.file.Path`, I think we could change the signature to accept `nio.file.Path` and return a collection of `nio.file.Path`.